custom/icds_reports/management/commands/nos_of_deaths.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import absolute_import, print_function
+
+from __future__ import unicode_literals
+import csv342 as csv
+import codecs
+import os
+
+from django.core.management.base import BaseCommand
+
+from corehq.apps.userreports.models import StaticDataSourceConfiguration
+from corehq.apps.userreports.util import get_indicator_adapter, get_table_name
+
+from corehq.sql_db.connections import connection_manager
+from io import open
+
+PERSON_TABLE_ID = 'static-person_cases_v2'
+AWC_LOCATION_TABLE_ID = 'static-awc_location'
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'file_path'
+        )
+
+    def handle(self, file_path, *args, **options):
+        domain = 'icds-cas'
+
+        data_source_id = StaticDataSourceConfiguration.get_doc_id(domain, PERSON_TABLE_ID)
+        config = StaticDataSourceConfiguration.by_id(data_source_id)
+        adapter = get_indicator_adapter(config)
+        session_helper = connection_manager.get_session_helper(adapter.engine_id)
+        person_table_name = get_table_name(domain, PERSON_TABLE_ID)
+        awc_location_table_name = get_table_name(domain, AWC_LOCATION_TABLE_ID)
+        session = session_helper.Session
+
+        with open(
+            os.path.join(os.path.dirname(__file__), 'sql_scripts', 'nos_of_deaths.sql'),
+            encoding='utf-8'
+        ) as f:
+            sql_script = f.read()
+            rows = session.execute(
+                sql_script % {
+                    'person_table_name': person_table_name,
+                    'awc_location_table_name': awc_location_table_name
+                }
+            )
+
+            utf_decoded_rows = []
```

Comment:
i dont think this part is necessary and is likely to break. If you are returning utf-8 encoded strings of non-ascii characters from the db, `str(elem)` will raise an error. Python should be able to handle the unicode just fine, and converting them to a string and then back to utf-8 does not change anything.

Reply:
I highly recommend watching and re-watching https://nedbatchelder.com/text/unipain.html

Reply:
While I was testing. It caused the problem writing data to file saying `\u2013` is not in ordinal range that is why I did this past and therefore the `codecs.open`

Reply:
Thanks Nick. Good post to understand how unicode. Finally made me understand what is encoding and decoding. and why required. Thanks